pkg/commands/convert/run.go

```diff
@@ -35,6 +37,7 @@ func Run(ctx context.Context, opts flag.Options) (err error) {
 		return xerrors.New("AWS and Kubernetes scanning reports are not yet supported")
 	}
 
+	compat(r)
```

Comment:
I'm confused. I see that `compat` function adds `UID`. Apparently this is due to the fact that slices are always pointers.

But I think we should explicitly make `Result` pointer.

Reply:
Yes, a report just holds pointers to a slice of results. We don't need to pass a pointer of the report here. If you get confused, we can change it.

Reply:
IIRC there may be problems with this approach if slice size changes
in this case the size doesn't change

but I think it’s better to choose the most obvious ways (without using the nuances of Go types), to avoid confusion when maintaining this code

Reply:
updated https://github.com/aquasecurity/trivy/pull/7042/commits/ff0ddfb4216a90a268ca5790ab08b6cb7de8d81f